mlrun/feature_store/feature_vector.py

```diff
@@ -308,10 +308,12 @@ def parse_features(self, offline=True, update_stats=False):
         index_keys = []
         feature_set_fields = collections.defaultdict(list)
         features = copy(self.spec.features)
+        label_column_name = None
         if offline and self.spec.label_feature:
             features.append(self.spec.label_feature)
             _, name, alias = parse_feature_string(self.spec.label_feature)
             self.status.label_column = alias or name
+            label_column_name = name
```

Comment:
You're only keeping the name of the feature, and discarding the feature-set name. This may cause an issue if there is a feature with the same name in another feature-set - that feature will get discarded even though it's not the label feature.
It should be perfectly valid to have a feature names `feat1` in both `fset1` and `fset2` and these features in the same fvec: `["fset1.*", "fset2.some_feature"]` and have `label_feature` to be `fset2.feat1`. When this happens, `fset1.feat1` should not be removed from the fvec (since it's not the label feature).
To enable that, you should probably keep here also the fset-name (which currently is discarded), and use that later in the comparison you're doing.